inox2d/src/render.rs

```diff
@@ -218,3 +222,173 @@ impl Puppet {
         }
     }
 }
+
+pub trait InoxRenderer
+where
+    Self: Sized,
+{
+    type Context;
+    type Error;
+
+    // create a new renderer, given rendering context
+    fn new(ctx: Self::Context) -> Result<Self, Self::Error>;
+
+    // do any model-specific setups, e.g. creating buffers with specific sizes
+    // after this step, the model provided should be renderable
+    fn prepare(&mut self, model: &Model) -> Result<(), Self::Error>;
+
+    // resize viewport
+    fn resize(&mut self, w: u32, h: u32) -> Result<(), Self::Error>;
```

Comment:
You can actually make all of these comments into documentation by using `///` instead of `//`. :D

Reply:
:D

(well, where is the official documentation on when and how to use `///` and `//`. I really don't know.